### PR TITLE
feat(toolbox): add plugin to show `toolbox` prompt info

### LINKED ISSUE #10446: Add prompt support for toolbx
### If the feature request is for a plugin or theme, specify it here.

_No response_

### If the feature solves a problem you have, specify it here.

[toolbx](https://github.com/containers/toolbox) is appending icon indicator to bash shell to indicate when your shell is running in container. When using oh-my-zsh this indicator is lost on every theme so there is no easy way to distinguish if you're running inside toolbx container or not.

### Describe the proposed feature.

Make use of `/run/.toolboxenv` file to determine if you're running inside toolbx shell and append toolbx ⬢ indicator to the beginning of PS1 independently of theme.
Allow to opt-out using flag in .zshrc.

### Describe alternatives you've considered

Modifying theme manually.

### Additional context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by carlosala**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

Adds `toolbox_prompt_info` function.
Closes #10446

### CODE REVIEW

File: plugins/toolbox/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# toolbox
+
+This plugins adds `toolbox_prompt_info()` function. It shows the toolbox indicator ⬢ if you are running in a container and nothing if not.
```
Comment (mcornella) [inline plugins/toolbox/README.md:3]: The first description line should avoid going into specifics. A better tagline would be "a prompt function for toolbox (insert link to project here)".

File: plugins/toolbox/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (mcornella) [inline plugins/toolbox/README.md:4]: Needs the usage standard info (to use it, add it ...).

File: plugins/toolbox/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+It can be useful to customize themes. For example:
```
Comment (mcornella) [inline plugins/toolbox/README.md:5]: Hmm I see you took the advice of using `kubectx` plugin as reference. That one's README should be fixed 😁 I like something more like the terraform plugin which is a little bit more verbose.

**Comment by mcornella** [comment]
@JayDoubleu would something like this work for you?

**Comment by JayDoubleu** [comment]
Hey @mcornella , apologies, I somehow managed to miss the notification. This works looks great!. I didn't know it would be so simple to implement ! 

I even prefer the indicator to be on the right side. Thank you.

Out of curiosity, is it possible to move it around, for example to the left side ?

**Comment by mcornella** [comment]
You could put the function in `$PS1` instead, in the position you prefer between the different segments of the prompt. If you want it all the way to the left, granted that your prompt is not multiple lines, you could just use
```sh
PS1='$(toolbox_prompt_info)'"$PS1"
```

We could document these two cases as the probable ones to be used by users.

**Comment by JayDoubleu** [comment]
I also recently found out tha there is `/run/.containerenv` available for toolbox users. Which returns bunch of metadata about the container. Perhaps an option to display the toolbox name alongside the icon would be a good idea `eval $(cat /run/.containerenv) && echo $name` .

I'll have a look at this tomorrow as it got me really excited ;)

**Comment by carlosala** [comment]
Now it should be better. Sorry for the shitty PR Marc ;)

**Comment by mcornella** [comment]
> I also recently found out tha there is `/run/.containerenv` available for toolbox users. Which returns bunch of metadata about the container. Perhaps an option to display the toolbox name alongside the icon would be a good idea `eval $(cat /run/.containerenv) && echo $name` .
> 
> I'll have a look at this tomorrow as it got me really excited ;)

Can you provide a sample .containerenv to see the format?

**Review by mcornella — APPROVED** [review]

**Comment by JayDoubleu** [comment]
> > I also recently found out tha there is `/run/.containerenv` available for toolbox users. Which returns bunch of metadata about the container. Perhaps an option to display the toolbox name alongside the icon would be a good idea `eval $(cat /run/.containerenv) && echo $name` .
> > I'll have a look at this tomorrow as it got me really excited ;)
> 
> Can you provide a sample .containerenv to see the format?

Sure thing it's 

```
cat /run/.containerenv 
engine="podman-3.4.4"
name="fedora-toolbox-35"
id="362b87172756af906fd8e4bd144fb6311baa0093ebbed7bbe7afdd6a9a851c41"
image="registry.fedoraproject.org/fedora-toolbox:35"
imageid="40b181c70b7342032df0b13c16beaf18a3ea7f73617858923c3972a02139d6e8"
rootless=1
```

Where name is the toolbox container name.

**Comment by carlosala** [comment]
I think we should merge this changes and open a new PR for more changes.

**Pull request merged**